PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationAdvancedPanel.java

```diff
@@ -0,0 +1,171 @@
+import java.awt.event.ItemEvent;
+import java.util.Arrays;
+import java.util.List;
+
+public class SqlServerCreationAdvancedPanel extends JPanel implements AzureFormPanel<SqlServerConfig> {
+
+    private JPanel rootPanel;
+    private ConnectionSecurityPanel security;
+    @Getter
+    private SubscriptionComboBox subscriptionComboBox;
+    @Getter
+    private ResourceGroupComboBox resourceGroupComboBox;
+    @Getter
+    private ServerNameTextField serverNameTextField;
+    @Getter
+    private SqlServerRegionComboBox regionComboBox;
+    @Getter
+    private AdminUsernameTextField adminUsernameTextField;
+    @Getter
+    private JPasswordField passwordField;
+    @Getter
+    private JPasswordField confirmPasswordField;
+
+    private AzurePasswordFieldInput passwordFieldInput;
+    private AzurePasswordFieldInput confirmPasswordFieldInput;
+
+    private SqlServerConfig config;
+
+    SqlServerCreationAdvancedPanel(SqlServerConfig config) {
+        super();
+        this.config = config;
+        $$$setupUI$$$(); // tell IntelliJ to call createUIComponents() here.
+        init();
+        initListeners();
+        setData(config);
+    }
+
+    private void init() {
+        passwordFieldInput = PasswordUtils.generatePasswordFieldInput(this.passwordField, this.adminUsernameTextField);
+        confirmPasswordFieldInput = PasswordUtils.generateConfirmPasswordFieldInput(this.confirmPasswordField, this.passwordField);
+    }
+
+    private void initListeners() {
+        this.subscriptionComboBox.addItemListener(this::onSubscriptionChanged);
+        this.adminUsernameTextField.getDocument().addDocumentListener(generateAdminUsernameListener());
+        this.security.getAllowAccessFromAzureServicesCheckBox().addItemListener(this::onSecurityAllowAccessFromAzureServicesCheckBoxChanged);
+        this.security.getAllowAccessFromLocalMachineCheckBox().addItemListener(this::onSecurityAllowAccessFromLocalMachineCheckBoxChanged);
+    }
+
+    private void onSubscriptionChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED && e.getItem() instanceof Subscription) {
+            final Subscription subscription = (Subscription) e.getItem();
+            this.resourceGroupComboBox.setSubscription(subscription);
+            this.serverNameTextField.setSubscription(subscription);
+            this.regionComboBox.setSubscription(subscription);
+        }
+    }
+
+    private DocumentListener generateAdminUsernameListener() {
+        return new TextDocumentListenerAdapter() {
+            @Override
+            public void onDocumentChanged() {
+                if (!adminUsernameTextField.isValueInitialized()) {
+                    adminUsernameTextField.setValueInitialized(true);
+                }
+            }
+        };
+    }
+
+    private void onSecurityAllowAccessFromAzureServicesCheckBoxChanged(final ItemEvent e) {
+        config.setAllowAccessFromAzureServices(e.getStateChange() == ItemEvent.SELECTED);
```

Comment:
As we have called `config.setAllowAccessFromAzureServices(security.getAllowAccessFromAzureServicesCheckBox().isSelected())` in `getData()`, do we need to add extra listener to change the config value?

Reply:
removing these lines is good and should be ok.